symplyphysics/definitions/mass_fraction_of_the_solute.py

```diff
@@ -0,0 +1,35 @@
+from sympy import (Eq, solve)
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
+    validate_output, dimensionless)
+
+# Description
+## The mass fraction is the ratio of the mass of the solute to the total mass of the solution
+## omega = m_i / m
+## Where:
+## m_i - mass of component
+## m - mass of solute
+## omega - mass fraction of the solute
+
+mass_fraction = Symbol("mass_fraction", dimensionless)
+mass_of_component = Symbol("mass_of_component", units.mass)
+mass_of_mixture = Symbol("mass_of_mixture", units.mass)
+
+definition = Eq(mass_fraction, mass_of_component / mass_of_mixture)
+
+definition_units_SI = dimensionless
+
+
+def print_law() -> str:
+    return print_expression(definition)
+
+
+@validate_input(mass_of_component_=mass_of_component, mass_of_mixture_=mass_of_mixture)
+@validate_output(mass_fraction)
+def calculate_mass_fraction(mass_of_component_: Quantity,
```

Comment:
I assume fraction is never higher than 100%. I suggest to create new `Fraction` type. You can use `Probability` as a reference.

Reply:
Like this?

```python
from typing import Self


class Fraction(float):
    
    def __new__(cls, value: float) -> Self:
        if value < 0 or value > 1.0:
            raise AttributeError(f"Fraction value should be in range [0..1], got '{value}'")
        return float.__new__(cls, value)
```

Reply:
Yes. It is copypaste though.

Reply:
We can inherit the 'Probability' class or vice versa, inherit the 'Probability' class from this class. But since the description of the 'AttributeError' error is changing, we have to recreate the `__new__()` method. Alternatively, the check for belonging to the interval [0, 1] should be placed in a separate class method, and then reused, while generating different error descriptions.

Reply:
For such trivial implementation it is better to copypaste and not worry about inheritance.

Reply:
Ok.